drivers: counter: nrfx_timer: Get TIMER max frequency from clock source

nordic-krch opened the pull request:
Some time ago `max-frequency` property was added to `nordic,nrf-timer` which indicated what is the clock frequency that runs the TIMER (it also indicates maximum frequency of the timer). Recently we started to add clock source to peripherals and frequency of the clock source shall be the source of information about the peripheral frequency.

PR includes:
- Revert  0ecfac663d4cb45 and 413ca65d65f9b4aceb13b4609d0141d9ee60e2f2
- Include work from #82871 which adds macro for getting frequency from the clock source
- Add missing clock sources to TIMER in nrf54x and nrf9280.

Fixes #83889.

nordic-krch committed:
Revert "dts: nordic nrf-timer: Expose max frequency as DT property"

This reverts commit 0ecfac663d4cb4593abee8e9353105f522ae49d2.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch committed:
dts: common: nordic: Add clock source to timers

Add clock source to timers which indicates maximum frequency of
the timer instance.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

masz-nordic commented on drivers/counter/counter_nrfx_timer.c:
```diff
@@ -419,14 +420,6 @@ static DEVICE_API(counter, counter_nrfx_driver_api) = {
 
-#if !defined(CONFIG_SOC_SERIES_BSIM_NRFXX)
-#define CHECK_MAX_FREQ(idx)									\
-		BUILD_ASSERT(DT_INST_PROP(idx, max_frequency) ==				\
-			NRF_TIMER_BASE_FREQUENCY_GET((NRF_TIMER_Type *)DT_INST_REG_ADDR(idx)))
```
This assert could still be used, just with the new macro.

nordic-krch committed:
boards: native: nrf_bsim: nrf54l15: Bring back clocks

Do not delete clocks for the bsim target. Clocks are referenced
by peripherals and it can be used to get frequency that clocks
the peripheral.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

anangl committed:
Revert "dts: common: nordic: nrf54l20: set timer frequency to 64MHz"

This reverts commit 413ca65d65f9b4aceb13b4609d0141d9ee60e2f2.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
- rebased
- replaced the workaround from 413ca65d65f9b4aceb13b4609d0141d9ee60e2f2 with a different solution to prevent a merge conflict

nordic-krch committed:
boards: nordic: Set 64M CPU frequency for nrf54l20pdk and nrf54l09pdk

nrf54l20pdk and nrf54l09pdk must use 64 MHz for now. So far it was
done by using SOC_NRF54LX_SKIP_CLOCK_CONFIG Kconfig option which was
skipping oscillator configuration so that it was running the default
frequency (which is 64 MHz). This approach was a bit cryptic because
DT was indicating that CPU was running 128 MHz when actual frequency
was different (and it was relying on assumption that default frequency
is 64 MHz).

After adding hfpll as clock source for CPU Kconfig option can
be replaced with DT setting where actual frequency is correctly
indicated. Since hfpll is a clock source for fast peripherals (e.g.
TIMER00) it is possible to have single source of information
regarding frequency.

Signed-off-by: Krzysztof Chruściński <krzysztof.chruscinski@nordicsemi.no>

nordic-krch commented:
@anangl @masz-nordic can you take a look? Currently counter tests are failing on some platforms because incorrect frequencies are read from DT and this PR fixes it.
